Sources/Rendering/Core/ImageCPRMapper/index.d.ts

```diff
@@ -0,0 +1,378 @@
+	uniformOrientation: TOrientation; // Don't use vec3 if possible
+	useUniformOrientation: boolean;
+	preferSizeOverAccuracy: boolean; // Whether to use halfFloat representation of float, when it is inaccurate
+	orientationArrayName: string | null;
+	tangentDirection: vec3;
+	bitangentDirection: vec3;
+	normalDirection: vec3;
+}
+
+export interface vtkImageCPRMapper extends vtkAbstractMapper3D {
+	/**
+	 * @returns the width of the image in model coordinates of the input volume
+	 */
+	getWidth(): number;
+
+	/**
+	 * @see getWidth
+	 * @param width
+	 */
+	setWidth(width: number): boolean;
+
+	/**
+	 * Use @see getUseUniformOrientation to use the this uniform orientation instead of the orientation specified by the centerline
+	 * @returns the uniform orientation of the centerline
+	 */
+	getUniformOrientation(): TOrientation;
+
+	/**
+	 * @see getUniformOrientation
+	 * @param orientation
+	 */
+	setUniformOrientation(orientation: TOrientation): boolean;
+
+	/**
+	 * This flag specifies wether the mapper should use the uniformOrientation ( @see getUniformOrientation ) or the orientation specified in centerline at centerline input ( @see setCenterlineData )
+	 * Defaults to false.
+	 * @returns the useUniformOrientation flag
+	 */
+	getUseUniformOrientation(): boolean;
+	
+	/**
+	 * @see getUseUniformOrientation
+	 * @param useUniformOrientation
+	 */
+	setUseUniformOrientation(useUniformOrientation: boolean): boolean;
+
+	/**
+	 * This flag indicates wether the GPU should use half float or not
+	 * When true, will use half float
+	 * When false, may use half float if there is no loss of accuracy (see in Texture: checkUseHalfFloat)
+	 * Defaults to false.
+	 * @returns the preferSizeOverAccuracy flag
+	 */
+	getPreferSizeOverAccuracy(): boolean;
+
+	/**
+	 * @see getPreferSizeOverAccuracy
+	 * @param preferSizeOverAccuracy
+	 */
+	setPreferSizeOverAccuracy(preferSizeOverAccuracy: boolean): boolean;
+
+	/**
+	 * OrientationArrayName specifies the name of the data array which gives an orientation for each point of the centerline
+	 * The data array has to be in the PointData attribute of the centerline input
+	 * If null, look for the orientation data array in: "Orientation", "Direction", Vectors, Tensors, Normals
+	 * The data array should be an array of mat4, mat3, quat or vec3 but using vec3 makes the CPRInteractor unusable
+	 * Default to null.
+	 */
+	getOrientationArrayName(): string | null;
+
+	/**
+	 * @see getOrientationArrayName
+	 * @param arrayName
+	 */
+	setOrientationArrayName(arrayName: string | null): boolean;
+
+	/**
+	 * For each point on the oriented centerline, the tangent direction is the direction in which the mapper will sample
+	 * Let O (a mat3) be the orientation at a point on a centerline, and N (a vec3) the tangent direction
+	 * Then the mapper will sample along O * N
+	 */
+	getTangentDirection(): vec3;
+
+	/**
+	 * @see getTangentDirection
+	 * @param tangent
+	 */
+	setTangentDirection(tangent: vec3): boolean;
```

Comment:
should this be renamed `setIJKTangentDirection` ?

Reply:
Why `IJK`?

Reply:
IJK is the local coordinate system of the volume (i.e. before applying volume `Direction` or actor `UserMatrix` )

Reply:
Then it is not the tangent direction in the IJK coordinate system
The tangent direction is in the oriented centerline coordinate system
To get the tangent direction in world coodinates:
WorldTangent = UserMatrix * CenterlineMatrix * TangentDirection
For example, if you want the tangent direction to follow the y axis of the centerline, set tangent direction to [0, 1, 0]